src/index.js

```
@@ -780,35 +778,50 @@ export default function (babel) {
             ]));
           }
           // may not be necessary?
-          path.get("body").scope.registerBinding("const", idPath);
+          // XXX: check with @rattrayalex re: what this is supposed to be doing.
+          // in the original source, this was registering the loop index
+          // binding in the loop scope. that's almost certainly not necessary.
+          // however, registering the newly created elem binding might be.
+          // no idea how this works tbh, can't make it compile, just commenting
+          // it out.
+          //path.get("body").scope.registerBinding("const", elemDecl);
```

Yeah, let's try without it. Will need to include a test that shadows it, expecting it to throw an error: 

```
for elem x in arr: 
  if true:
    x = 3
```
should throw an error. 

If it does, this isn't needed, I think